exporters/common/src/main/java/io/opentelemetry/exporter/grpc/GrpcSenderConfig.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.grpc;
+
+import io.opentelemetry.exporter.compressor.Compressor;
+import io.opentelemetry.sdk.common.export.RetryPolicy;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.X509TrustManager;
+
+/**
+ * Configuration for {@link GrpcSender} implementations, provided via {@link
+ * GrpcSenderProvider#createSender(GrpcSenderConfig)}.
+ */
+@Immutable
+public interface GrpcSenderConfig {
+
+  /**
+   * The gRPC endpoint to send to, including scheme. Omits path, which must be constructed from
+   * {@link #getFullServiceName()} and {@link #getMethodName()}.
+   */
+  URI getEndpoint();
+
+  /**
+   * The fully qualified gRPC service name, e.g. {@code
+   * opentelemetry.proto.collector.trace.v1.TraceService}.
+   */
+  String getFullServiceName();
+
+  /** The gRPC method name, e.g. {@code Export}. */
+  String getMethodName();
+
+  /**
+   * The compressor, or {@code null} if no compression is used. If present, {@link
+   * Compressor#compress(OutputStream)} must be applied to {@link
+   * GrpcMessageWriter#writeMessage(OutputStream)} when {@link GrpcSender#send(GrpcMessageWriter,
+   * Consumer, Consumer)} is called and {@link Compressor#getEncoding()} must be set as the {@code
+   * grpc-encoding}.
+   */
+  @Nullable
+  Compressor getCompressor();
```

Comment:
At some point I'll need to create Compressor objects but the existing implementations are in internal packages.

Reply:
This PR promotes them to the stable package. The idea is that this config option gives you a ref to the resolved compressor, or `null` if none is configured. Hopefully the quarkus sender can integrate with the compressor, adding the appropriate headers and using it to compress the payload.